### PR TITLE
Added support for string, int(8,16,32,64), uint(8,16,32,64) and each …

### PR DESCRIPTION
**Pull request by aren55555**
…of their ptr types as acceptable to use for the ID field.

### CODE REVIEW

File: request.go [hunk 1]
```diff
@@ -163,24 +163,93 @@ func unmarshalNode(data *Node, model reflect.Value, included *map[string]*Node)
...
+
+			// Deal with PTRS
+			var kind reflect.Kind
+			if fieldValue.Kind() == reflect.Ptr {
+				kind = fieldType.Type.Elem().Kind()
+			} else {
+				kind = fieldType.Type.Kind()
+			}
+
+			var idErr error
```
Comment (shwoodard) [inline request.go]: we don't need another `error`, use `err`.

File: request.go [hunk 2]
```diff
@@ -163,24 +163,93 @@ func unmarshalNode(data *Node, model reflect.Value, included *map[string]*Node)
...
+			switch kind {
+			case reflect.String:
```
Comment (shwoodard) [inline request.go]: don't use nested switch. use if then break out if str.

Comment (aren55555) [inline request.go]: Done!

File: request.go (hunk 1 above)
Comment (aren55555) [inline request.go]: Done!

**Pull request merged**